components/camel-spring-ldap/src/main/java/org/apache/camel/component/springldap/SpringLdapProducer.java

```diff
@@ -89,12 +91,19 @@ public void process(Exchange exchange) throws Exception {
             throw new UnsupportedOperationException("LDAP operation must not be empty, but you provided an empty operation");
         }
 
+        LdapTemplate ldapTemplate = endpoint.getLdapTemplate();
+
         String dn = (String)body.get(DN);
+        if (StringUtils.isBlank(dn)) {
+            ContextSource contextSource = ldapTemplate.getContextSource();

```

Comment:
The exchange feels more local. I think it makes sense to consider the exchange DN as an override.

Would you explain your thinking or how you would use it where preferring the context source's base DN makes sense?